rom/dev/tools/test-fmc/src/main.rs

```diff
@@ -195,32 +195,44 @@ fn get_pcr_entry(entry_index: usize) -> PcrLogEntry {
     PcrLogEntry::read_from_prefix(pcr_entry.as_bytes()).unwrap()
 }
 
-fn process_mailbox_commands() {
-    let mut mbox = unsafe { caliptra_registers::mbox::MboxCsr::new() };
-    let mbox = mbox.regs_mut();
-
+fn process_mailbox_command(mbox: &caliptra_registers::mbox::RegisterBlock<RealMmioMut>) {
     let cmd = mbox.cmd().read();
     cprintln!("[fmc] Received command: 0x{:08X}", cmd);
     match cmd {
         0x1000_0000 => {
-            read_pcr_log(&mbox);
+            read_pcr_log(mbox);
         }
         0x1000_0001 => {
-            create_certs(&mbox);
+            create_certs(mbox);
         }
         0x1000_0002 => {
-            read_fuse_log(&mbox);
+            read_fuse_log(mbox);
         }
         0x1000_0003 => {
-            read_fht(&mbox);
+            read_fht(mbox);
         }
         0x1000_0004 => {
-            trigger_update_reset(&mbox);
+            trigger_update_reset(mbox);
         }
         _ => {}
     }
 }
 
+fn process_mailbox_commands() {
+    let mut mbox = unsafe { caliptra_registers::mbox::MboxCsr::new() };
+    let mbox = mbox.regs_mut();
+
+    #[cfg(feature = "interactive_test_fmc")]
+    loop {
+        if mbox.status().read().mbox_fsm_ps().mbox_execute_uc() {
+            process_mailbox_command(&mbox);
+        }
+    }
+
+    #[cfg(not(feature = "interactive_test_fmc"))]
+    process_mailbox_command(&mbox);
```

Comment:
Is this necessary? It should be fine to just always use the interactive version since we aren't booting an RT image anyway.

The test can just run 1 or more mailbox commands and then exit.

Reply:
I initially tried this but then had to touch several other tests that expect test-fmc to run and quit without interaction.

Reply:
Ya, other tests would need to be changed, just think it ends up with a cleaner final product. But I don't feel too strongly about this.

Reply:
Not just the integration tests, there's also a GitHub presubmit that invokes ROM makefile's `run` target, which assembles test-fmc with test-rt and doesn't provide any handy way to invoke a host command to get the image to actually exit.